### PR TITLE
Toolbar: fix header width recalculation (T752661, T752306, T752400)

### PR DESCRIPTION
**Pull request by viterobk**

### CODE REVIEW

File: testing/tests/DevExpress.ui/themes.tests.js [hunk 1]
```diff
@@ -568,17 +568,6 @@ QUnit.module("web font checker", () => {
         assert.equal(diff, 0, "Element's count are the same after method call");
     });
 
-    test("waitWebFont: function reject by timeout if the font is not loaded", (assert) => {
```
Comment (babich-a) [inline testing/tests/DevExpress.ui/themes.tests.js:571]: I suggest rewriting this test to make sure that promise will be resolved if the font is not loaded.

File: js/ui/toolbar/ui.toolbar.base.js [hunk 2]
```diff
@@ -149,12 +149,38 @@ var ToolbarBase = AsyncCollectionWidget.inherit({
...
+    },
+
     _render: function() {
+        var that = this;
         this.callBase();
         this._renderItemsAsync();
 
         if(themes.isMaterial()) {
-            this._checkWebFontForLabelsLoaded().then(this._dimensionChanged.bind(this));
+            this._waitParentAnimationFinished().then(function() {
```
Comment (babich-a) [inline js/ui/toolbar/ui.toolbar.base.js]: We can use an arrow function. We do not need 'that' in this case. My suggestion is `Promise.all([this._waitParentAnimationFinished(),this._checkWebFontForLabelsLoaded()]).then(this._dimensionChanged.bind(this))`

File: js/ui/toolbar/ui.toolbar.base.js [hunk 3]
```diff
@@ -149,12 +149,38 @@ var ToolbarBase = AsyncCollectionWidget.inherit({
...
     },
 
+    _waitParentAnimationFinished: function() {
+        var $element = this.$element(),
+            timeout = 15;
+        return new Promise(function(resolve) {
+            const check = function() {
+                var animated = true;
+                $element.parents().each(function(_, parent) {
+                    if($(parent).data("dxAnimData")) {
```
Comment (babich-a) [inline js/ui/toolbar/ui.toolbar.base.js]: If somebody rename 'dxAnimData' constant the trouble will happen. I suggest using https://js.devexpress.com/Documentation/ApiReference/Common/utils/fx/Methods/#isAnimatingelement method from 'fx' that do completely the same thing.

File: js/ui/toolbar/ui.toolbar.base.js [hunk 4]
```diff
@@ -149,12 +149,38 @@ var ToolbarBase = AsyncCollectionWidget.inherit({
         this.setAria("role", "toolbar");
     },
 
+    _waitParentAnimationFinished: function() {
+        var $element = this.$element(),
+            timeout = 15;
+        return new Promise(function(resolve) {
```
Comment (babich-a) [inline js/ui/toolbar/ui.toolbar.base.js]: Will modern syntax be better `... Promise(resolve => { ...` ?

File: js/ui/toolbar/ui.toolbar.base.js [hunk 5]
```diff
@@ -149,12 +149,38 @@ var ToolbarBase = AsyncCollectionWidget.inherit({
         this.setAria("role", "toolbar");
     },
 
+    _waitParentAnimationFinished: function() {
+        var $element = this.$element(),
```
Comment (babich-a) [inline js/ui/toolbar/ui.toolbar.base.js]: const/let instead of var?

**Review by babich-a — CHANGES_REQUESTED** [review]

**Review by babich-a — APPROVED** [review]

**Review by BingoRUS — APPROVED** [review]

**Comment by hakimio** [comment]
@viterobk 
Using the latest build (19.1.5-build-19191-0321) the issue doesn't seem to be fixed. 
Demo -> https://codesandbox.io/s/popup-editing-devextreme-data-grid-rzpy4
Click the edit button to open edit dialog.

**Comment by viterobk** [comment]
Hello @hakimio 
Thank you for the feedback. We reproduced this issue and [fixed](https://github.com/DevExpress/DevExtreme/pull/8903) it.

**Pull request merged**